### DISCUSSION TITLE
Apply texture to lines

### DISCUSSION THREAD

**Discussion by Shonik** [pelicanmapping/osgearth#1798] [Support Q&A] [answered]
Hello,

I would like to apply a texture to a LineString geometry.

To achieve this, I tried to apply a style with the imageURI property pointing to my texture with no success.
Here is the code I used : 

```c++

mGeometry = new LineString();
mFeature = new Feature(mGeometry, geoSRS);
mFeature->geoInterp() = GEOINTERP_GREAT_CIRCLE;
setFeature(mFeature);

_style.getOrCreate<RenderSymbol>()->depthOffset()->enabled() = true;
_style.getOrCreate<LineSymbol>()->stroke()->smooth() = true;
_style.getOrCreate<LineSymbol>()->tessellationSize()->set(75000, Units::METERS);
_style.getOrCreate<LineSymbol>()->color() = Color::White;
_style.getOrCreate<LineSymbol>()->width() = 100;
_style.getOrCreate<LineSymbol>()->widthUnits() = Units::METERS;
_style.getOrCreate<LineSymbol>()->imageURI() = StringExpression("C:/Users/Public/CUBE/git/Core/Qml/Components/CoreProperties/OSGTextureProperties/images/zoneWeatherConflict.png");
_style.getOrCreate<AltitudeSymbol>()->clamping() = AltitudeSymbol::CLAMP_ABSOLUTE;

```
The lines appear just white, with no texture.
Could you help me figure out what is wrong ?

Thanks

**Comment by gwaldron** [comment]
This shouldn't even compile, since you need
```c++
_style.getOrCreate<LineSymbol>()->stroke()->color() = Color::White;
_style.getOrCreate<LineSymbol>()->stroke()->width() = 100;
_style.getOrCreate<LineSymbol>()->stroke()->widthUnits() = Units::METERS;
```
Assuming the png file exists, it's possible that the shader is not installed properly. Is this a FeatureNode? Is it under the MapNode in the scene graph?

**Comment by Shonik** [comment]
Yes you're right, my bad.
I went too fast when copy/pasting. My code is written exactly like you said.

Indeed, this happens in the constructor of a FeatureNode.
It is then added to an osg::Group which is added as a child of the MapNode.

**Comment by gwaldron** [comment, marked as answer]
Thanks. I reproduced the problem and pushed a fix for it. This was a bug in the BuildGeometryFilter class that prevented the imageURI from working on the FeatureNode (or anything besides the FeatureModelLayer). 

Other issue exist with imageURI (since it is kind of an experimental feature) but this should at least solve your immediate issue.

**Reply by Shonik** [reply]
Thanks, it's working perfectly with the fix ! 

Would it be possible to specify the line width in pixels and still have the texture working, or is it a limitation ?

**Reply by gwaldron** [reply]
That's a limitation for now. Pixel width lines use a totally different code path (LineDrawable) so there would be some development work involved. If you want you can open a GItHub ticket and we will mark it as a feature request. Thanks!

**Reply by Shonik** [reply]
OK thanks, I submitted a ticket.
